compute/ARMComputeEx/src/core/CL/kernels/CLSplitVKernelEx.cpp

```diff
@@ -0,0 +1,248 @@
+  uint32_t reversed = (input->info()->num_dimensions() - *split_dim - 1);
+  if (layout_diff < 0)
+  {
+    if (reversed == 0)
+    {
+      return 2;
+    }
+    if (reversed == 1)
+    {
+      return 0;
+    }
+    if (reversed == 2)
+    {
+      return 1;
+    }
+  }
+  else if (layout_diff > 0)
+  {
+    if (reversed == 0)
+    {
+      return 1;
+    }
+    if (reversed == 1)
+    {
+      return 2;
+    }
+    if (reversed == 2)
+    {
+      return 0;
+    }
+  }
+
+  return reversed;
+}
+
+Status validate_slices(const ITensorInfo *input, const std::vector<ITensorInfo *> &outputs,
+                       uint32_t split_dim)
+{
+//   ARM_COMPUTE_RETURN_ERROR_ON_NULLPTR(input);
+//   ARM_COMPUTE_RETURN_ERROR_ON(split_dim >= input->num_dimensions());
+//   ARM_COMPUTE_RETURN_ERROR_ON(outputs.size() < 2);
+
+  // Start/End coordinates
+  Coordinates start_coords;
+  Coordinates end_coords;
+  for (unsigned int d = 0; d < input->num_dimensions(); ++d)
+  {
+    end_coords.set(d, -1);
+  }
+  unsigned int axis_offset = 0;
+  // Validate output tensors
+  for (const auto &output : outputs)
+  {
+    // ARM_COMPUTE_RETURN_ERROR_ON_NULLPTR(output);
+    // Get output shape
+    const TensorShape output_shape = output->tensor_shape();
+    // ARM_COMPUTE_RETURN_ERROR_ON(output_shape.total_size() == 0);
+
+    const size_t axis_split_step = output_shape[split_dim];
+
+    // Output auto inizialitation if not yet initialized
+    TensorInfo tmp_output_info = *output->clone();
+    auto_init_if_empty(tmp_output_info,
+                       input->clone()->set_is_resizable(true).set_tensor_shape(output_shape));
+
+    // Update coordinate on axis
+    start_coords.set(split_dim, axis_offset);
+    end_coords.set(split_dim, axis_offset + axis_split_step);
+
+    // ARM_COMPUTE_RETURN_ON_ERROR(CLSlice::validate(input, output, start_coords, end_coords));
+    Status s = CLSlice::validate(input, output, start_coords, end_coords);
+    if (bool(s))
+    {
+        std::cout << "CLSlice::validate failed with status " << std::to_string(s.error_code) << std::endl;
+    }
+    
+    axis_offset += axis_split_step;
+  }
+
+  return Status{};
+}
+
+void configure_slices(const ICLTensor *input, const std::vector<ICLTensor *> &outputs,
+                      std::vector<CLSlice> &_slice_functions, uint32_t split_dim_revised)
+{
+  unsigned int axis_offset = 0;
+  // Start/End coordinates
+  Coordinates start_coords;
+  Coordinates end_coords;
+  for (unsigned int d = 0; d < input->info()->num_dimensions(); ++d)
+  {
+    end_coords.set(d, -1);
+  }
+  int out_iter = 0;
+  for (const auto &output : outputs)
+  {
+    const TensorShape output_shape = output->info()->tensor_shape();
+    auto op_size = output_shape.total_size();
+    if (!op_size)
+    {
+      continue;
+    }
+    // assert(out_iter < 6);
+    assert(op_size != 0);
+    assert(split_dim_revised >= output_shape.num_dimensions());
```

Comment:
`assert(split_dim_revised < output_shape.num_dimensions());` ?

Reply:
It should be reversed, thanks for spotting.